File: javaQuicktheories/src/main/java/properties/JsonSerializer.java
package properties;

import java.io.IOException;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

public class JsonSerializer {
	public static String toString(Person p) throws JsonProcessingException {
		ObjectMapper mapper = new ObjectMapper();
		return mapper.writeValueAsString(p);
	}
	
	public static Person fromString(String s) throws IOException {
		ObjectMapper mapper = new ObjectMapper();
		return mapper.readValue(s, Person.class);
	}
}

File: kotlin/src/main/kotlin/properties/JsonSerializer.kt
package properties

import com.fasterxml.jackson.databind.ObjectMapper

data class Address(var street: String, var town: String, var zip: Int) {
    constructor(): this("","",0)
}

data class Person(var name: String, var address: Address) {
    constructor(): this("",Address())
}

object JsonSerializer {
    fun toString(p: Person): String = ObjectMapper().writeValueAsString(p)

    fun fromString(s: String): Person = ObjectMapper().readValue(s, Person::class.java)
}
